dropwizard-jackson/src/main/java/io/dropwizard/jackson/FuzzyEnumModule.java

```diff
@@ -48,22 +50,36 @@ protected PermissiveEnumDeserializer(Class<Enum<?>> clazz) {
                     .removeFrom(jp.getText())
                     .replace('-', '_')
                     .replace('.', '_');
+
             for (Enum<?> constant : constants) {
                 if (constant.name().equalsIgnoreCase(text)) {
                     return constant;
                 }
             }
 
+            //In some cases there are certain enums that don't follow the same patter across an enterprise.  So this
+            //means that you have a mix of enums that use toString(), some use @JsonCreator, and some just use the
+            //standard constant name().  This block handles finding the proper enum by toString()
+            String toStringText = jp.getText()
```

Comment:
Should we really replace hyphens and underscores, if we want deserialize on the result of the `toString` method?
It could return a string with these symbols. For example, something like that:
```java
private enum CurrencyCode {
    USD("united_states_dollar"),
    EUR("euro"),
    GBP("pound_sterling");

    private final String description;

    CurrencyCode(String name) {
        this.description = name;
    }

    @Override
    public String toString() {
        return description;
    }
}
```